docs/core/versions/binding.md

```diff
@@ -0,0 +1,117 @@
+
+If no acceptable `2.0.*` version is found, a new `2.*` version will be used. For example, if you specified `netcoreapp2.0` and only `2.1.0` is installed, the application will run using the `2.1.0` runtime. This This behavior is referred to as "minor version roll-forward." Lower versions also won't be considered. When no acceptable runtime is installed, the application will not run.
+
+A few usage examples demonstrate the behavior:
+
+- 2.0.4 is required. 2.0.5 is the highest patch version available. 2.0.5 is used.
+- 2.0.4 is required. No 2.0.* versions are installed. 1.1.1 is the highest runtime installed. An error message is printed.
+- 2.0.4 is required. No 2.0.* versions are installed. 2.2.2 is the highest 2.x runtime version installed. 2.2.2 is used.
+- 2.0.4 is required. No 2.x versions are installed. 3.0.0 is installed. An error message is printed.
+
+Minor version roll-forward has one side-effect that may affect end users. Consider the following scenario:
+
+- 2.0.4 is required. No 2.0.* versions are installed. 2.2.2 is installed. 2.2.2 is used.
+- 2.0.5 is later installed. 2.0.5 will be used for subsequent application launches, not 2.2.2.
+- It's possible that 2.0.5 and 2.2.2 might behave differently, particularly for scenarios like serializing binary data.
+
+All NuGet dependencies are resolved as part of the publish operation and reside as assemblies (.dlls) in a flat directory structure. These assemblies should run on .NET Core 2.1 due to its compatibility promise for existing applications and binaries.
+
+Developers should remember the following rules:
+
+- Runtimes are searched for in the following order (the first found is loaded):
+  - Highest patch version (check that the minimum runtime patch version is satisfied, otherwise error)
+  - Highest minor version
+- If an acceptable runtime is not found, the application does not run and an error message explains why.
+
+## Self-contained deployments include the selected runtime
+
+You can publish an application as a [**self-contained distribution**](../deploying/index.md#self-contained-deployments-scd). This approach includes .NET Core. Self-contained distributions do not have a dependency on runtime environments. Runtime binding occurs at publishing time, not run-time.
+
+The publishing process selects the latest patch version of the given runtime family. For example, `dotnet publish` will select .NET Core 2.0.4 if it is the latest patch version in the .NET Core 2.0 runtime family. The target framework (including the latest installed security patches) are packaged with the application.
+
+It is an error if the minimum version specified for an application is not satisfied. `dotnet publish` binds to latest runtime patch version (within a given major.minor version family). `dotnet publish` does not support the roll-forward semantics of `dotnet run`.
```

Comment:
I'd suggest linking from here to https://docs.microsoft.com/en-us/dotnet/core/deploying/runtime-patch-selection, which includes more details on roll-forward for self-contained publish.